epochStart/bootstrap/startInEpochScheduled.go

```diff
@@ -343,16 +343,22 @@ func isScheduledIntermediateTx(
 	return isScheduledIntermediateTransaction && isTxExecutedInSelfShard
 }
 
-func getScheduledIntermediateTxsMap(
+func getScheduledIntermediateTxsMapInOrder(
 	miniBlocks map[string]*block.MiniBlock,
 	intermediateTxs map[string]data.TransactionHandler,
 ) map[block.Type][]data.TransactionHandler {
 
 	intermediateTxsMap := make(map[block.Type][]data.TransactionHandler)
 
-	for txHash, tx := range intermediateTxs {
-		blockType := getBlockTypeOfTx([]byte(txHash), miniBlocks)
-		intermediateTxsMap[blockType] = append(intermediateTxsMap[blockType], tx)
+	for _, miniBlock := range miniBlocks {
```

Comment:
:+1:

Reply:
isn't here an issue, due to range on the miniBlocks which is as well a map?
in case of multiple mbs of the same type, the constructed intermediateTxsMap may hold the mbs intermediate txs sets in a different order.